plugins/Lb302/Lb302.cpp

```diff
@@ -585,20 +585,23 @@ int Lb302Synth::process(sampleFrame *outbuf, const int size)
 				vco_k = 0.5 * Oscillator::noiseSample( vco_c );
 				break;
 
+			// The next cases all use the BandLimitedWave class which uses the oscillator increment `vco_inc` to compute samples.
+			// If that oscillator increment is 0 we return a 0 sample because calling BandLimitedWave::pdToLen(0) leads to a
+			// division by 0 which in turn leads to floating point exceptions.
 			case VcoShape::BLSawtooth:
-				vco_k = BandLimitedWave::oscillate( vco_c + 0.5f, BandLimitedWave::pdToLen( vco_inc ), BandLimitedWave::Waveform::BLSaw ) * 0.5f;
+				vco_k = vco_inc == 0. ? 0. : BandLimitedWave::oscillate( vco_c + 0.5f, BandLimitedWave::pdToLen( vco_inc ), BandLimitedWave::Waveform::BLSaw ) * 0.5f;
```

Comment:
We should use something like `typeInfo<float>::isEqual` when comparing floating point values by equality. The formatting (whitespace) could also be fixed too.

Also, instead of repeating the same condition (which will become more obvious if you add `typeInfo<float>::isEqual`), why not just protect against 0 for the entire switch case using an `if` statement?

Reply:
> We should use something like `typeInfo<float>::isEqual` when comparing floating point values by equality. The formatting (whitespace) could also be fixed too.

I only want to protect against a division by 0. `typeInfo<float>::isEqual` uses an epsilon area which is not wanted in this case.

> Also, instead of repeating the same condition (which will become more obvious if you add `typeInfo<float>::isEqual`), why not just protect against 0 for the entire switch case using an `if` statement?

There are several cases in the switch statement that do not take `vco_inc` into account at all, i.e. which are implemented differently. If I would put the check for 0 around the switch statement then these cases wouldn't have any output whenever `vco_inc` is zero although that variable is not related to their implementation.

The underlying problem is that the code is very spaghetti and mixed and should be made more modular. However, I do not want to touch that as I only want to get rid of the divisions by 0.